internal/xds/rbac/rbac_engine_test.go

```diff
@@ -0,0 +1,625 @@
+		},
+		{
+			name: "TestCertainHeaders",
+			rbacConfig: &v3rbacpb.RBAC{
+				Policies: map[string]*v3rbacpb.Policy{
+					"certain-headers": {
+						Permissions: []*v3rbacpb.Permission{
+							{Rule: &v3rbacpb.Permission_Any{Any: true}},
+						},
+						Principals: []*v3rbacpb.Principal{
+							{
+								Identifier: &v3rbacpb.Principal_OrIds{OrIds: &v3rbacpb.Principal_Set{Ids: []*v3rbacpb.Principal{
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ExactMatch{ExactMatch: "GET"}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_SafeRegexMatch{SafeRegexMatch: &v3matcherpb.RegexMatcher{Regex: "GET"}}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_RangeMatch{RangeMatch: &v3typepb.Int64Range{
+										Start: 0,
+										End:   64,
+									}}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_PresentMatch{PresentMatch: true}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_PrefixMatch{PrefixMatch: "GET"}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_SuffixMatch{SuffixMatch: "GET"}}}},
+									{Identifier: &v3rbacpb.Principal_Header{Header: &v3routepb.HeaderMatcher{Name: ":method", HeaderMatchSpecifier: &v3routepb.HeaderMatcher_ContainsMatch{ContainsMatch: "GET"}}}},
+								}}},
+							},
+						},
+					},
+				},
+			},
+			successfullyConstruct: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			_, err := NewEngine(test.rbacConfig)
+			if test.successfullyConstruct {
+				// If the config is supposed to successfully construct, then
+				// there should be no error raised.
+				if err != nil {
+					t.Fatalf("RBAC Engine should have successfully constructed and not returned an error")
+				}
+			} else {
+				// If the config has a logical problem, then there should be an
+				// error returned.
+				if err == nil {
+					t.Fatalf("RBAC Engine should not have constructed and there should have been an error returned")
+				}
+			}
+		})
+	}
+}
+
+// TestRBACEngine tests the RBAC Engine by configuring the engine in different
+// scenarios. After configuring the engine in a certain way, this test pings the
+// engine with different kinds of data representing incoming RPC's, and verifies
+// that it works as expected.
+func (s) TestRBACEngine(t *testing.T) {
+	tests := []struct {
+		name        string
+		rbacConfig  *v3rbacpb.RBAC
+		rbacQueries []struct {
+			evaluateArgs           *RPCData
```

Comment:
`evaluateArgs` needs to be renamed at this point. `rpcData` should be fine.

Reply:
Switched.